build-logic/src/main/kotlin/kotlinx/io/conventions/kotlinx-io-multiplatform.gradle.kts

```diff
@@ -30,17 +31,11 @@ kotlin {
     }
 
     jvm {
-        withJava()
         testRuns["test"].executionTask.configure {
             useJUnitPlatform()
         }
-        // can be replaced with just `compilerOptions { }` in Kotlin 2.0
-        compilations.configureEach {
-            compileTaskProvider.configure {
-                compilerOptions {
-                    freeCompilerArgs.add("-Xjvm-default=all")
-                }
-            }
+        compilerOptions {
+            jvmDefault = JvmDefaultMode.NO_COMPATIBILITY
```

Comment:
https://kotlinlang.org/docs/whatsnew22.html#changes-to-default-method-generation-for-interface-functions